src/yaml/yaml-network-parser.c

```diff
@@ -334,16 +383,17 @@ int parse_yaml_network_file(const char *file, Network **ret) {
 
                 done = !yaml_document_get_root_node(&document);
                 if (!done)
-                        r = parse_yaml_document(m, &document, network);
+                        r = parse_yaml_document(m, &document, network, link);
 
                 yaml_document_delete(&document);
         }
 
         yaml_parser_delete(&parser);
 
-        if (r >= 0)
-                *ret = steal_pointer(network);
-
+        if (r >= 0) {
+                *n = steal_pointer(network);
+                *l = steal_pointer(link);
```

Comment:
hmm either this will be a network or link. Both can be same together. We need to take care of it